### PR TITLE
[4.0] Change com_associations to services

### PR DESCRIPTION
**Pull request by laoneo**
Moves com_associations to the service based extension setup.

See #20217 for details.

### CODE REVIEW

**Comment by infograf768** [comment]
Tested: No noticeable change after this PR (I mean same issues as before).
(To test, one has first to patch with https://github.com/joomla/joomla-cms/pull/20448 )

**Comment by laoneo** [comment]
Thanks for the test.

**Comment by wilsonge** [comment]
Can we/should we have a library class for the component classes that are just MVCFactory aware? Because we're going to have com_cpanel, com_modules, com_plugins, com_associations etc all having exactly the same classname, which seems kinda pointless (to be clear a generic version of https://github.com/Digital-Peak/joomla-cms/blob/9c95d4ece19fe25697fcecce8cb98c25d8b71d8a/administrator/components/com_associations/Extension/AssociationsComponent.php)

**Comment by laoneo** [comment]
@wilsonge good input. Changed and could remove two more classes.

File: libraries/src/Extension/MVCComponent.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+use Joomla\CMS\MVC\Factory\MVCFactoryServiceTrait;
+
+/**
+ * MVC Component class.
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class MVCComponent extends Component implements MVCFactoryServiceInterface
+{
+	use MVCFactoryServiceTrait;
```
Comment (laoneo) [inline libraries/src/Extension/MVCComponent.php:23]: Do we still want to keep the trait as it is used only here anymore?

Comment (wilsonge) [reply libraries/src/Extension/MVCComponent.php:23]: I think so - people might still choose to build up their own hierarchy of classes for their component implementations

**Review by wilsonge — APPROVED** [review]
LGTM - need to solve the conflicts here from the requirement of the dispatcher class.

**Comment by laoneo** [comment]
Synced

**Pull request merged**